packages/syft/src/syft/core/tensor/autograd/tensor.py

```diff
@@ -2,138 +2,138 @@
-        autograd.backward_ops.AddOp,
-        autograd.backward_ops.SubOp,
-        autograd.backward_ops.MulOp,
-        autograd.backward_ops.DotOp,
-    ]:
+    ) -> Optional[Type[autograd.backward_ops.Op]]:
         if not self.requires_grad:
             raise Exception("This tensor is not backpropagated")
         return self._grad_fn
 
     # Autograd Tensor Operations
+    """ Note: Ignoring return type incompatibilities since AutogradTensorAncestor doesn't inherit from
+    PassThroughTensor"""
+
     def __abs__(self) -> AutogradTensorAncestor:
         op = autograd.backward_ops.AbsOp()
         return op(self)
 
-    def __add__(self, other: AutogradTensor) -> AutogradTensorAncestor:
+    def __add__(self, other: AutogradTensor) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.AddOp()
         return op(self, other)
 
-    def __sub__(self, other: AutogradTensor) -> AutogradTensorAncestor:
+    def __sub__(self, other: AutogradTensor) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.SubOp()
         return op(self, other)
 
-    def __mul__(self, other: AutogradTensor) -> AutogradTensorAncestor:
+    def __mul__(self, other: AutogradTensor) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.MulOp()
         return op(self, other)
 
-    def __rmul__(self, other: AutogradTensor) -> AutogradTensorAncestor:
+    def __rmul__(self, other: AutogradTensor) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.MulOp()
         return op(self, other)
 
-    def __truediv__(self, other: AutogradTensor) -> AutogradTensorAncestor:
+    def __truediv__(self, other: AutogradTensor) -> AutogradTensorAncestor:  # type: ignore
         if is_acceptable_simple_type(other):
-            return self * (1 / other)
+            # Ignoring type annotation error because only int, floats, np.ndarrays will be parsed
+            return self * (1 / other)  # type: ignore
         return NotImplemented
 
-    def __pow__(
-        self, other: Union[AutogradTensor, Union[int, bool, float, Any]]
-    ) -> AutogradTensorAncestor:
+    def __pow__(self, other: Any) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.PowOp()
         return op(self, other)
 
-    def __rpow__(
-        self, other: Union[AutogradTensor, Union[int, bool, float, Any]]
-    ) -> AutogradTensorAncestor:
+    def __rpow__(self, other: Any) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.RPowOp()
         return op(self, other)
 
-    def reshape(self, *shape: tuple) -> AutogradTensorAncestor:
+    def reshape(self, *shape: tuple) -> AutogradTensorAncestor:  # type: ignore
         op = autograd.backward_ops.ReshapeOp()
         return op(self, *shape)
 
-    def repeat(self, *args: int, **kwargs: int) -> AutogradTensorAncestor:
+    def repeat(self, *args: int, **kwargs: int) -> AutogradTensorAncestor:  # type: ignore
```

Comment:
args and kwargs are not int, change to Tuple[Any, ...] and Any

Reply:
fixed